packages/upswyng-web/src/components/__tests__/Resource.test.tsx

```diff
@@ -25,27 +32,18 @@ jest.mock("../../App.styles", () => ({
     },
   },
 }));
-jest.mock("react-router-dom", () => ({
-  useParams: () => ({ resourceId: "some resource ID" }),
-}));
-jest.mock("../PageBanner", () => "PageBanner");
-jest.mock("../Schedule", () => "Schedule");
-jest.mock("../Map", () => "Map");
-jest.mock("../useResource.tsx");
-const mockedUseResource = useResource as jest.Mock;
 
-describe("<Resource/>", () => {
-  mockedUseResource.mockImplementation(() => foodResource) as typeof jest.mock;
-  const wrapper = shallow(<Resource />);
+const mockUseResource = useResource as jest.Mock;
 
-  it("renders the charityname property of the resource prop object", () => {
-    expect(wrapper.find("PageBanner").prop("text")).toBe(foodResource.name);
-  });
+describe("<Resource/>", () => {
+  const setup = (overrides = {}) => {
+    return mount(<Resource {...overrides} />);
+  };
 
   it("renders a loading spinner when a resource is not loaded", () => {
-    mockedUseResource.mockImplementation(() => undefined) as typeof jest.mock;
-    const wrapper = shallow(<Resource />);
+    mockUseResource.mockReturnValueOnce(undefined);
+    const wrapper = setup();
 
-    expect(wrapper.find("LoadingSpinner").length).toBe(1);
+    expect(wrapper.text()).toBe("TEST-loading-spinner");
```

Comment:
With the updated mock of the loading spinner, we're now only looking for text (checking via `.text()`) rather than an actual component, which we look for with `.find()`